### PR TITLE
fix: #557 Configure CLIENT API Token For DEMO

### PR DESCRIPTION
**Pull request by ianliuwk1019**
Configuring "**DEMO**" deployment to use CLIENT "**PROD**" for batch data refreshing.

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-8.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-8.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-8.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

### CODE REVIEW

File: .github/workflows/merge-demo.yml [hunk 1]
```diff
@@ -69,6 +69,7 @@ jobs:
               -p DB_TESTDATA=true
               -p AWS_USER_POOLS_WEB_CLIENT_ID="k3b9ip1vf85o4tkqvu5g4adgj"
               -p LOGOUT_CHAIN_URL="https://logontest7.gov.bc.ca/clp-cgi/logoff.cgi?retnow=1&returl=https://test.loginproxy.gov.bc.ca/auth/realms/standard/protocol/openid-connect/logout?redirect_uri="
+              -p BATCH_CLIENT_RFSH_API_TKN_OP_SECRET_NAME="fom-demo-client-app-api"
```
Comment (ianliuwk1019) [inline .github/workflows/merge-demo.yml:72]: OpenShift YAML template does not seem to support conditional statement(IF/ELSE).
I did see AZURE pipeline template allows that.
So I had to instead using "Parameter" replacement in template for getting the correct secret object "name" from OpenShift based on where it is merged from (`merge-demo.yml`).

**Review by MCatherine1994 — APPROVED** [review]

**Review by basilv — COMMENTED** [review]
looks fine.

**Comment by basilv** [comment]
Weird, I'm trying to approve the PR but can't.

**Comment by basilv** [comment]
(I don't have the option in GitHub to approve the PR.)

**Comment by ianliuwk1019** [comment]
> (I don't have the option in GitHub to approve the PR.)

Oh, sorry @basilv I just saw this. The PR is merged and closed yesterday. Thanks for looking at this, I thought you were very busy so Catherine approved it and I merged it and closed it.

**Pull request merged**